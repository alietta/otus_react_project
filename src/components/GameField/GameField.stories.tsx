import React from "react";
import { withKnobs, number, array, object } from "@storybook/addon-knobs";
import { GameField } from "./GameField";

export default {
  title: "GameField",
  decorators: [withKnobs],
};

export const fieldWithCells = () => (
  <GameField
    xSize={number("xSize", 3)}
    ySize={number("ySize", 3)}
    field={object("field", [
      [false, true, false],
      [false, true, false],
      [false, true, false],
    ])}
  />
);
