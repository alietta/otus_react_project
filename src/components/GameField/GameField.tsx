import React, { FunctionComponent, useState } from 'react';

interface GameFildProps {
  xSize: number;
  ySize: number;
  field: string[][],
}

const GameField: FunctionComponent<GameFildProps> = (props) => {
  const [ field, setField ] = useState<Array<boolean>>(props.field);
  return <div>field</div>;
};

export { GameField };
