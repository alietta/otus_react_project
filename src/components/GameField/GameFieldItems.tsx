import styled from "@emotion/styled";
import { css } from "@emotion/core";

export const Field = styled.div`
  box-sizing: border-box;
  border: 3px solid gray;
  padding: 3px;
  display: inline-block;
`;

export const Row = styled.div`
  display: flex;
  align-items: flex-start;
`;
