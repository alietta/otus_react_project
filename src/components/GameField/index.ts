export * from "./GameField";
export * from "./components";
