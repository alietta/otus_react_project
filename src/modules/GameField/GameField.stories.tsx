import React, { FC } from "react";
import { withKnobs, object } from "@storybook/addon-knobs";
import { GameField } from "./GameField";

export default {
  title: "GameField",
  decorators: [withKnobs],
};

export const fieldWithCells: FC = () => (
  <GameField
    field={object("field", [
      [false, true, false],
      [false, true, false],
      [false, true, false],
    ])}
  />
);
