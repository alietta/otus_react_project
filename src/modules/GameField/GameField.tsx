import React, { FunctionComponent, useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";
import { useTheme } from "sancho";
import { Field } from "./GameFieldItems";
import { Cell } from "components/Cell";
import { actions } from "./duck/reducer";
import { makeField } from "../Game/gameFunctions";

interface GameFildProps {
  cellSize: number;
}

const GameField: FunctionComponent<GameFildProps> = (props: GameFildProps) => {
  const theme = useTheme();
  const dispatch = useDispatch();
  const { cellSize } = props;
  const field = useSelector((state: any) => state.field);
  const filledCells = useSelector((state: any) => state.game.filledCells);
  const fieldSize = useSelector((state: any) => state.settings.fieldSize);
  const onClick = (): void => {
    console.log("click");
  };

  useEffect(() => {
    const newField = makeField(fieldSize.width, fieldSize.height, filledCells);
    dispatch(actions.setField(newField));
  }, [filledCells]);

  return (
    <Field
      css={{
        borderColor: theme.colors.border.default,
      }}
    >
      {field.map((arr, index) => {
        return (
          <div key={`row-${index}`} css={{ display: "flex" }}>
            {arr.map((isFilled, ind) => {
              return (
                <Cell
                  key={`y-${index}_x-${ind}`}
                  isFilled={isFilled}
                  x={ind}
                  y={index}
                  onClick={onClick}
                  size={cellSize}
                />
              );
            })}
          </div>
        );
      })}
    </Field>
  );
};

export { GameField };
