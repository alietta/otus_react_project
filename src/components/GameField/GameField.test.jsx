/**
 * @jest-environment jsdom
 */
import React from "react";
import { mount, configure } from "enzyme";
import renderer from "react-test-renderer";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

import { GameField } from "./GameField";

describe("GameField", () => {
  it("renders cells for passed empty GameField", () => {
    expect(
      renderer
        .create(
          <GameField
            field={[
              [false, false],
              [false, false],
            ]}
          />
        )
        .toJSON()
    ).toMatchSnapshot();
  });
  it("renders filled cells snapshot check", () => {
    expect(
      renderer
        .create(
          <GameField
            field={[
              [true, false],
              [false, false],
            ]}
          />
        )
        .toJSON()
    ).toMatchSnapshot();
  });
});
