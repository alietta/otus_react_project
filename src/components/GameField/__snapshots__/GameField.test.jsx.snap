// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameField renders cells for passed empty GameField 1`] = `
<div
  className="css-fwl64t-Field enkp98f0"
>
  <div>
    <div
      className="css-6scvsp-CellWrapper-EmptyCell ej7ht1y0"
      onClick={[Function]}
    />
    <div
      className="css-6scvsp-CellWrapper-EmptyCell ej7ht1y0"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="css-6scvsp-CellWrapper-EmptyCell ej7ht1y0"
      onClick={[Function]}
    />
    <div
      className="css-6scvsp-CellWrapper-EmptyCell ej7ht1y0"
      onClick={[Function]}
    />
  </div>
</div>
`;

exports[`GameField renders filled cells snapshot check 1`] = `
<div
  className="css-fwl64t-Field enkp98f0"
>
  <div>
    <div
      className="css-1imi224-CellWrapper-FilledCell ej7ht1y0"
      onClick={[Function]}
    />
    <div
      className="css-6scvsp-CellWrapper-EmptyCell ej7ht1y0"
      onClick={[Function]}
    />
  </div>
  <div>
    <div
      className="css-6scvsp-CellWrapper-EmptyCell ej7ht1y0"
      onClick={[Function]}
    />
    <div
      className="css-6scvsp-CellWrapper-EmptyCell ej7ht1y0"
      onClick={[Function]}
    />
  </div>
</div>
`;
